samcli/commands/local/lib/sam_api_provider.py

```diff
@@ -86,6 +89,41 @@ def _extract_from_serverless_api(self, logical_id, api_resource, collector, cwd=
         self.extract_swagger_api(logical_id, body, uri, binary_media, collector, cwd)
         collector.add_stage_name(logical_id, stage_name)
         collector.add_stage_variables(logical_id, stage_variables)
+        if cors:
+            collector.add_cors(logical_id, cors)
+
+    def _extract_cors(self, properties):
+        """
+        Extract Cors property from AWS::Serverless::Api resource by reading and parsing Swagger documents. The result
+        is added to the Api.
+
+        Parameters
+        ----------
+        properties : dict
+            Resource properties
+        """
+        cors_prop = properties.get("Cors")
+        cors = None
+        if cors_prop and isinstance(cors_prop, dict):
+            allow_methods = cors_prop.get("AllowMethods", ','.join(AbstractApiProvider.ANY_HTTP_METHODS))
+
+            if allow_methods and "OPTIONS" not in allow_methods:
+                allow_methods += ",OPTIONS"
+
+            cors = Cors(
+                allow_origin=cors_prop.get("AllowOrigin"),
+                allow_methods=allow_methods,
+                allow_headers=cors_prop.get("AllowHeaders"),
+                max_age=cors_prop.get("MaxAge")
+            )
+        elif cors_prop and isinstance(cors_prop, string_types):
```

Comment:
If cors in not a dict, then it must be a string right? we dont accept anything else, so do we need this if else?

Reply:
The other case is the default case of no cors.